### DISCUSSION TITLE
How can I detect serverless function's timeout in my Node.js project

### DISCUSSION THREAD

**Discussion by Tupiet** [vercel/vercel#8589] [Help] [answered]
# UPDATE: I can't delete this discussion
I've seen that this repository's discussions is only about Vercel CLI and that I should ask it in the community repository, so [that's what I did now](https://github.com/vercel/community/discussions/882). But I cannot delete this discussion from here. Sorry. :(

---

# ORIGINAL:

Hey! Some months ago, I deployed a Node.js project to Vercel, using the serverless functions. The project consists of an API (express.js), which returns me some data generated randomly. The user makes a request to this API and send a JSON, which contains the type of data (names, emails, numbers, etc.) and the quantity of data. And here's the problem: as I'm in the hobby plan, the timeout for the serverless functions is about 10 seconds. Depending on the quantity, the timeout is exceeded or not, firing a 504 error. 

I'd like the API to send a message if the timeout was exceeded, but I don't know how I can do it. This is the code of the API, which gets the body of the request (the data I want to receive, with JSON format), then that JSON is managed, and finally I send another JSON with the data:

```
app.post('/api', (req, res) => {
    const body = req.body

    let result = manageData(body)
    res.send(result)
})
```

Is there any way I could detect if I'm exceeding the timeout, and then send a message explaining the problem?

Hope anyone can help me, and thanks for reading this!

If you need more data, please tell me. 

 Have a nice day.

**Comment by swarnava** [comment, marked as answer]
Hi,

If you have setup Logdrain with your application, you can create an alert when execution reach to certain threshold. Learn more here: https://vercel.com/support/articles/how-do-i-store-logs-on-vercel
